### PR TITLE
HSEARCH-3850 adding builder in BackendSettings & IndexSettings

### PR DESCRIPTION
**Pull request by acucumel**

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/engine/cfg/IndexSettings.java [hunk 1]
```diff
@@ -19,4 +21,39 @@
...
+	 * </p>
+	 *
+	 * @param backendName Expect the backendName
+	 * @param luceneOrElasticsearchBackendSettings Expect one of ElasticsearchIndexSettings constants string
+	 *
+	 * @return the concatenated default index settings key
+	 */
+	public static String indexDefaultsKey(String backendName, String luceneOrElasticsearchBackendSettings) {
+		return join( ".",
+				EngineSettings.BACKENDS, backendName, BackendSettings.INDEX_DEFAULTS, luceneOrElasticsearchBackendSettings
```
[the remark this answers is not in the document]
Comment (acucumel) [reply engine/src/main/java/org/hibernate/search/engine/cfg/IndexSettings.java]: sounds good to me

File: documentation/src/main/asciidoc/configuration.asciidoc [hunk 2]
```diff
@@ -119,6 +119,31 @@ or a reference by type as a `java.lang.Class`
...
+Both `BackendSettings` and `IndexSettings` provide tools to help build the configuration property keys.
+
+
+BackendSettings::
+`BackendSettings.backendKey("myBackend", ElasticsearchBackendSettings.HOSTS)` is equivalent to `hibernate.search.backends.myBackend.host`
+link:{hibernateSearchJavadocUrl}/org/hibernate/search/backend/elasticsearch/cfg/ElasticsearchBackendSettings.html[ElasticsearchBackendSettings]
+link:{hibernateSearchJavadocUrl}/org/hibernate/search/backend/lucene/cfg/LuceneBackendSettings.html[LuceneBackendSettings]
+
+IndexSettings::
+`IndexSettings.indexDefaultsKey("myBackend", ElasticsearchIndexSettings.LIFECYCLE_MINIMAL_REQUIRED_STATUS)` is equivalent to `hibernate.search.backends.signature.index_defaults.lifecycle.minimal_required_status`
```
[the remark this answers is not in the document]
Comment (acucumel) [reply documentation/src/main/asciidoc/configuration.asciidoc]: yes, i forgot this part, thank you

File: documentation/src/test/java/org/hibernate/search/documentation/configuration/MyConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import org.hibernate.search.backend.elasticsearch.cfg.ElasticsearchIndexSettings;
+import org.hibernate.search.backend.elasticsearch.index.IndexLifecycleStrategyName;
+import org.hibernate.search.backend.elasticsearch.multitenancy.MultiTenancyStrategyName;
+import org.hibernate.search.engine.cfg.BackendSettings;
+import org.hibernate.search.engine.cfg.EngineSettings;
+import org.hibernate.search.engine.cfg.IndexSettings;
+import org.hibernate.search.mapper.orm.automaticindexing.session.AutomaticIndexingSynchronizationStrategyNames;
+import org.hibernate.search.mapper.orm.cfg.HibernateOrmMapperSettings;
+
+public class MyConfiguration {
```
[the remark this answers is not in the document]
Comment (acucumel) [reply documentation/src/test/java/org/hibernate/search/documentation/configuration/MyConfiguration.java]: @yrodiere 
```
@Test
public void shouldBuildHibernateConfiguration() {
	assertThat( buildHibernateConfiguration())
		.containsExactly(
			entry( "hibernate.search.backends.myBackend.hosts", "127.0.0.1:9200" ),
			entry( "hibernate.search.backends.myBackend.protocol", "http" )
	);
}
```
Are you ok with something like that ?

**Comment by acucumel** [comment]
> Looks good to me! I just squashed your commits and pushed an additional commit to try and solve the problem with the test not being taken into account when computing test coverage. Let's see how it goes, then I'll merge.

thx, i look into the code covering and i was wondering the same thing lol

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**